core/src/core-api/_internalLinkManager.js

```diff
@@ -43,10 +45,12 @@ export class linkManager extends LuigiCoreAPIBase {
         modal: modalSettings,
         splitView: splitViewSettings,
         drawer: drawerSettings
-      })
+      }),
+      remotePromiseId: remotePromise.id
     };
 
     this.sendPostMessageToLuigiCore(navigationOpenMsg);
+    return remotePromise;
   }
 
   openAsModal(path, modalSettings = {}) {
```

Comment:
I would assume open as modal, drawer, splitview should also return a promise.
If it is too much work also fine in follow-up task. If so could you create this task?

Reply:
I will create new Task for this and implement it in this task.